### PR TITLE
feat: `PasswordInput` component redesign

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
## Description

- Adds `size` prop

## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

**Review by anuraghazra — APPROVED** [review]

File: packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx
Comment (saurabhdaware) [inline packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx:1]: The dots look a tiny in new input. Is intended on design?
image 
image

Comment (saurabhdaware) [reply packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx:1]: Can you check the label left story once? the helpText shifts in it
image

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx:1]: Fixed this

Comment (saurabhdaware) [reply packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx:1]: https://github.com/razorpay/blade/pull/2050#discussion_r1510681367 This one is expected?

**Review by saurabhdaware — APPROVED** [review]
Small [doubt](https://github.com/razorpay/blade/pull/2050#discussion_r1510681367). rest looks good to me

**Pull request merged**